### PR TITLE
add fcm for android

### PR DESCRIPTION
**Pull request by mysticeti**
Added FCM notification for android. Web support is not added yet.
To see notifications through FCM, first, open the app and go to the FCM page. Press the get token button to see the unique token for the device. Copy the token and paste it into the cloud messaging of the Firebase console Send a test message to view the notification message in the device.

### CODE REVIEW

File: web/index.html [hunk 1]
```diff
@@ -51,6 +51,8 @@
   <script src="https://www.gstatic.com/firebasejs/8.4.1/firebase-auth.js"></script>
   <script src="https://www.gstatic.com/firebasejs/8.4.1/firebase-firestore.js"></script>
   <script src="https://www.gstatic.com/firebasejs/8.4.1/firebase-functions.js"></script>
+  <script src="https://www.gstatic.com/firebasejs/8.4.1/firebase-messaging.js"></script>
+
```
Comment (joykuotw) [inline web/index.html:55]: This is for the web. Add it in other PR.

File: lib/main.dart [hunk 2]
```diff
@@ -30,21 +33,143 @@ import 'home_page.dart';
...
+              android: initializationSettingsAndroid,
+              iOS: initializationSettingsIOS);
+
+      Future<void> initiateFlutterLocalNotificationPlugin() async {
+        await flutterLocalNotificationsPlugin.initialize(initializationSettings,
+            onSelectNotification: selectNotification);
+      }
+
+      initiateFlutterLocalNotificationPlugin();
+    }
```
Comment (joykuotw) [inline lib/main.dart:133]: I believe these are all unnecessary. If you remove the codes and test again, the messaging should work fine.

**Review by joykuotw — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 2 above)
Comment (mysticeti) [reply lib/main.dart:133]: These codes allow jumping to the demo screen when the notification is pressed when the app is open. selectNotification does the switching. If I do not include the above code then, only when the app is closed/in the background, pressing the notification causes the screen to change.

File: web/index.html (hunk 1 above)
Comment (mysticeti) [reply web/index.html:55]: Removed in the next commit

File: lib/main.dart (hunk 2 above)
Comment (mysticeti) [reply lib/main.dart:133]: I wanted to show that when a notification pops up depending on the data, users can be navigated to that screen as well.

**Review by joykuotw — APPROVED** [review]

**Pull request merged**